app.js

```diff
@@ -345,6 +372,7 @@ var eventsModule = function () {
                 efDataHolder[templateName].frames.find(ef=>ef.id === result).attributeValuesMap = attributeMap;
             }
         }
+        displayCallback();
```

Comment:
Yeah, i don't think having this call back here is great, but it seems necessary unless we restructure data calls a bit.

i think we can start by adjusting the `GetAttributesValues` function to automatically `ProcessAttributeResults` and then return the whole thing in a promise.